src/Fixer/Operator/BinaryOperatorSpacesFixer.php

```diff
@@ -695,6 +701,22 @@ private function injectAlignmentPlaceholdersForArrow(Tokens $tokens, int $startA
                     ++$index;
                 }
             }
+
+            if ($token->equals('{')) {
+                $until = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_CURLY_BRACE, $index);
+                $this->injectArrayAlignmentPlaceholders($tokens, $index + 1, $until - 1);
+                $index = $until;
+
+                continue;
+            }
+
+            if ($token->equals('(')) {
```

Comment:
Since I stopped to ignore the if/foreach/... skip, I need to add a check to create a new scope every time a (, or a { is open.

This way the two `=>` are not considered in the same scope.
```
foreach ($a => $b) {
    foreach ($c => $d) {
}}
```

This strategy was already used for the alignement of all the others operators.